File: Queues/BFSQueue.java
import java.util.LinkedList;
import java.util.Queue;

public class BFSQueue {

    static int[][] adj = {
        {0, 1, 1, 0, 0, 0},
        {1, 0, 0, 1, 1, 0},
        {1, 0, 0, 0, 1, 0},
        {0, 1, 0, 0, 0, 1},
        {0, 1, 1, 0, 0, 1},
        {0, 0, 0, 1, 1, 0}
    };

    static int V = 6;

    static void bfs(int start) {
        boolean[] visited = new boolean[V];
        Queue<Integer> queue = new LinkedList<>();

        visited[start] = true;
        queue.add(start);

        System.out.print("BFS Traversal: ");

        while (!queue.isEmpty()) {
            int v = queue.poll();
            System.out.print(v + " ");

            for (int i = 0; i < V; i++) {
                if (adj[v][i] == 1 && !visited[i]) {
                    visited[i] = true;
                    queue.add(i);
                }
            }
        }
    }

    public static void main(String[] args) {
        bfs(0);   // Starting BFS from vertex 0
    }
}

File: Queues/bfs_queue.py
from collections import deque

adj = [
    [0, 1, 1, 0, 0, 0],
    [1, 0, 0, 1, 1, 0],
    [1, 0, 0, 0, 1, 0],
    [0, 1, 0, 0, 0, 1],
    [0, 1, 1, 0, 0, 1],
    [0, 0, 0, 1, 1, 0]
]

V = 6

def bfs(start):
    visited = [False] * V
    queue = deque()

    visited[start] = True
    queue.append(start)

    print("BFS Traversal:", end=" ")

    while queue:
        v = queue.popleft()
        print(v, end=" ")

        for i in range(V):
            if adj[v][i] == 1 and not visited[i]:
                visited[i] = True
                queue.append(i)

bfs(0)   # Starting BFS from vertex 0
